kaitai/kaitaistream.h

```diff
@@ -4,13 +4,14 @@
 // Kaitai Struct runtime API version: x.y.z = 'xxxyyyzzz' decimal
 #define KAITAI_STRUCT_VERSION 11000L
 
-#include <istream>
-#include <sstream>
-#include <stdint.h>
-#include <sys/types.h>
-#include <limits>
-#include <stdexcept>
-#include <errno.h>
+#include <stdint.h> // int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t
+
+#include <ios> // std::streamsize
+#include <istream> // std::istream
+#include <limits> // std::numeric_limits
+#include <sstream> // std::istringstream
+#include <string> // std::string
+#include <type_traits> // std::enable_if, std::is_integral
```

Comment:
This one seems to be C++11 and above only. Probably worth guarding with ifdefs.

Reply:
Yeah, I'm glad our CI caught this nicely because I hadn't thought to test it locally.

[**linux (98)**](https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/actions/runs/8887186835/job/24402033190#step:4:37) (gcc 11.4.0 with `-std=c++98`)

```
In file included from /usr/include/c++/11/type_traits:35,
                 from /home/runner/work/kaitai_struct_cpp_stl_runtime/kaitai_struct_cpp_stl_runtime/kaitai/kaitaistream.h:14,
                 from /home/runner/work/kaitai_struct_cpp_stl_runtime/kaitai_struct_cpp_stl_runtime/kaitai/kaitaistream.cpp:1:
/usr/include/c++/11/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
   32 | #error This file requires compiler and library support \
      |  ^~~~~
```

[**linux_ubuntu_7_10**](https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/actions/runs/8887186835/job/24402033367#step:6:7) (gcc 4.1)

```
In file included from kaitai/kaitaistream.cpp:1:
./kaitai/kaitaistream.h:14:59: error: type_traits: No such file or directory
```

Reply:
Resolved in https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/pull/72/commits/69cb210ca353c309ec18bf75b025c44da6f5210b